File: Data Structures & Algorithms/duplicate-integer/submission-0.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        Map<Integer, Integer> seen = new HashMap<>();
        int i = 0;
        for (int num : nums) {
            if (!seen.containsKey(num)) {
                seen.put(num, i);
                i++;
            }
            else{ 
                return true;
            }
        }
    return false;
    }
}

File: Data Structures & Algorithms/search-2d-matrix/submission-0.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        
        for nums in matrix:
            for i in nums:
                l,r = 0, len(nums) - 1
                if nums[r] < target:
                    break
                while l <= r:
                    m = l + ((r - l) // 2)
                    if nums[m] > target:
                        r = m - 1
                    elif nums[m] < target:
                        l = m + 1
                    if nums[m] == target:
                        return True
        return False

                